src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/ProjectConfigurationFileChangeEventArgs.cs

```diff
@@ -32,7 +33,7 @@ internal sealed class ProjectConfigurationFileChangeEventArgs : EventArgs
         _gate = new object();
     }
 
-    public bool TryDeserialize([NotNullWhen(true)] out RazorProjectInfo? projectInfo)
+    public bool TryDeserialize(LanguageServerFeatureOptions languageServerFeatureOptions, [NotNullWhen(true)] out RazorProjectInfo? projectInfo)
```

Comment:
Leaving as draft because I'm not particularly happy with this design, but it represents the best of "After this method is called, you have what best represents our understanding of state". 

The other option is for callers to modify, but that also seemed messy

Reply:
I think without the change to `RazorLanguageFeatureFlags` this PR would actually be pretty clean. Would be worth some comments saying that the flags are specifically controlled by the server and _not_ serialized, but I think the idea works

Reply:
Ah good call. I'll add a comment on the class around the intent

Reply:
This just got me thinking of "separation" and I might be overly separating too early

Reply:
This system would only apply to things that flow to the compiler, right? Whilst I can't think of any others we might want, a) you've already done the work, b) you did it in separate commits so its not too bad, and c) having a single property that represents "these things aren't serialized and are set by the language server" is probably better than X individual properties with that behaviour.